### ISSUE TITLE
Regression: Version 3 does not merge path and operation properties

### ISSUE THREAD

**Issue by kevinoid** [swagger-api/swagger-js#965]
In a specification where parameters are defined on both the path and the operation, the parameters from the path appear to be ignored.  For example:

```js
'use strict';

const assert = require('assert');
const http = require('http');
const SwaggerClient = require('swagger-client');

const TEST_HOST = 'localhost';
const TEST_ID = 123;
const TEST_PORT = 8888;

http
  .createServer((req, res) => {
    assert.strictEqual(req.url, `/cars/${TEST_ID}`);
    res.writeHead(200, {'Content-Type': 'application/json'});
    res.end('{}');
  })
  .listen(TEST_PORT, TEST_HOST, () => {
    new SwaggerClient({
      usePromise: true,
      spec: {
        swagger: '2.0',
        info: {
          title: 'Path Parameter API',
          version: '1.0.0'
        },
        host: `${TEST_HOST}:${TEST_PORT}`,
        schemes: [
          'http'
        ],
        paths: {
          '/cars/{carId}': {
            parameters: [
              {
                name: 'carId',
                in: 'path',
                type: 'number',
                required: true
              }
            ],
            get: {
              operationId: 'getCar',
              parameters: [
                {
                  name: 'clean',
                  in: 'query',
                  type: 'boolean'
                }
              ],
              responses: {
                200: {
                  description: 'A car',
                  schema: {
                    type: 'object'
                  }
                }
              }
            }
          }
        }
      }
    })
      .then(client => client.apis.default.getCar({carId: TEST_ID}))
      .then(
        () => {
          console.log('It works.');
          process.exit();
        },
        err => {
          console.error(err);
          process.exit(1);
        }
      );
  });
```

With `swagger-js` version 3.0.1 this fails with `AssertionError: '/cars/%7BcarId%7D' === '/cars/123'` while `swagger-js` version 2.1.32 prints `It works.`.  I bisected the regression to 7678d40, but that's obviously not very helpful since almost everything was replaced in that commit.

Thanks,
Kevin

**Comment by webron** [comment]
@kevinoid Thanks for the report. This is a complete rewrite so previous code may not be relevant. I was sure we had a test for this case, but we'll definitely look into it.

**Comment by kevinoid** [comment]
Great, thanks @webron!

**Comment by webron** [comment]
Thanks for reporting the issue. Please report new ones as you find them. If you can help us with writing tests and even fixes PRs, that would be appreciated. We hope to formalize a process for it later this week.

**Comment by kevinoid** [comment]
Will do.  So far this is the only issue I've noticed.

I actually started writing a test for this one but got bogged down when the test behaved differently than calling the API directly (looks like `parameters` is moved out of the `path` object somewhere outside of the `buildRequest` function used in the tests, but I could be wrong on that).  I'll try to send a test and/or fix with future issues.

**Comment by saharj** [comment]
I'm still working on this and trying to solve the issue.
The interesting thing I've found is that if you put path-parameters after `get` like below:
```
paths: {
  '/cars/{carId}': {
    get: {
      operationId: 'getCar',
      parameters: [
        {
          name: 'clean',
          in: 'query',
          type: 'boolean'
        }
      ],
      responses: {
        200: {
          description: 'A car',
          schema: {
            type: 'object'
          }
        }
      }
    },
    parameters: [
      {
        name: 'carId',
        in: 'path',
        type: 'number',
        required: true
      }
    ]
  }
}
```
the URL will have the `{carId}` in it, but the query parameter `clean` won't get included in the URL.

**Comment by kevinoid** [comment]
Thanks @saharj and @buunguyen!  The fix is working well for me with v3.0.3.
